### ISSUE TITLE
Problems on Ubuntu 16.04 (with Python 3.5)

### ISSUE THREAD

**Issue by mherrmann3** [obspy/obspy#1387]
Hi there,

I recently updated my Ubuntu to the most recent version 16.04 (Xenial) and get the following error message when importing any obspy module in Python 3.5.1+:

```
ImportError: Could not load shared library "libmseed_Linux_64bit_py35.cpython-35m-x86_64-linux-gnu.so".

 /usr/lib/python3/dist-packages/obspy/core/util/../../lib/libmseed_Linux_64bit_py35.cpython-35m-x86_64-linux-gnu.so: cannot open shared object file: No such file or directory
```
Indeed, only `libmseed_Linux_64bit_py34.cpython-34m.so` exists there.

I reinstalled the python3-obspy package for the trusty release (since it's the highest release for which you offer the python3 version of obspy: [1.0.1-1~trusty](http://deb.obspy.org/pool/main/p/python-obspy)), but the error persists. Importing an obspy modul in python 2.7 works, though.
What also works: obspy in an anaconda3 environment (Python 3.5.1 :: Anaconda 4.0.0); indeed, the `libmseed_Linux_64bit_py35.cpython-35m-x86_64-linux-gnu.so` exists.

I have no problem in using the anaconda environment, but was wondering how to fix the issue using the classic approach (apt repository). Why are there differences b/w apt and anaconda? Can you provide a new Debian package for python 3.5?

Many thanks for your help!

**Comment by megies** [comment]
> I recently updated my Ubuntu to the most recent version 16.04 (Xenial) and get the following error message when importing any obspy module in Python 3.5.1+:

How did you install to get this particular error?

> I have no problem in using the anaconda environment, but was wondering how to fix the issue using the classic approach (apt repository). Why are there differences b/w apt and anaconda? Can you provide a new Debian package for python 3.5?

I didn't get around to setting up the build environment for the new release yet, will try to do it soonish.. in any case, trying to install a package built for an older release is bound to fail.

**Comment by mherrmann3** [comment]
> How did you install to get this particular error?

I did an upgrade from Ubuntu 15.04 (vivid) to 15.10, then 16.04, then reinstalled obspy:
`sudo apt-get install --reinstall python-obspy python3-obspy`

> I didn't get around to setting up the build environment for the new Ubuntu release 16.04 yet, will try to do it soonish

Thanks in advance! (speaking for all up-to-date ubuntu users) :stuck_out_tongue_closed_eyes: 

> in any case, trying to install a package built for an older release is bound to fail.

I imagined, but worth a try :wink:

**Comment by megies** [comment]
> I did an upgrade from Ubuntu 15.04 (vivid) to 15.10, then 16.04, then reinstalled obspy:
> sudo apt-get install --reinstall python-obspy python3-obspy

Ok, I think you might still have some misdirected repository URLs in your `/etc/apt/sources.list`, mentioning an old condename e.g. `trusty`. Otherwise you wouldn't have been able to get apt to install ObsPy (as there's no `xenial` package as of today).

**Comment by mherrmann3** [comment]
Yes, that is correct. I particularly did that for obspy since Ubuntu would otherwise not be able to install (or find) python3-obspy at all. That it doesn't work now is another story :wink:
As [mentioned in the link above](http://deb.obspy.org/pool/main/p/python-obspy/), _trusty_ is your most recent package for python3; for python2 it's _wheezy_. 
I have them both installed, but only the latter works.

**Comment by megies** [comment]
> for python2 it's wheezy

Nope, there's also trusty packages for Python2. See http://deb.obspy.org/pool/main/p/python-obspy/.

**Comment by mherrmann3** [comment]
Yes, you're right. Wheezy is Debian - I thought its a newer Ubuntu :confused:. However, I have it installed on my Ubuntu (it did it automatically) and it works :smiley:

**Comment by megies** [comment]
Ubuntu 16.04 `xenial` packages are up now, [test look fine](http://tests.obspy.org/?node=xenial_amd64). Happy to hear feedback on the packages. Closing issue, please reopen if there are any problems with the packages.

**Comment by mherrmann3** [comment]
Yesss, it works. Perfect! Thanks a lot, Tobias. Btw: nice testing interface :open_mouth:

**Comment by lijunzh** [comment]
> Ubuntu 16.04 `xenial` packages are up now

@megies I am wondering how did you fix that particular error. 

After a recent `sudo apt upgrade`, I start to see the similar error message as following: `ImportError: Could not load shared library "libmseed_Linux_64bit_py37.cpython-@PYVERNODOTS@m-x86_64-linux-gnu.so".`

I installed the `obspy` via the conda-forge channel. My `conda list` looks as following:

```
# packages in environment at /home/lijun/conda:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main
asn1crypto                0.24.0                   py37_0
backcall                  0.1.0                    py37_0
blas                      1.0                         mkl
bzip2                     1.0.8                h7b6447c_0
ca-certificates           2019.6.16            hecc5488_0    conda-forge
certifi                   2019.6.16                py37_1
cffi                      1.12.3           py37h2e261b9_0
chardet                   3.0.4                 py37_1003
conda                     4.7.11                   py37_0    conda-forge
conda-package-handling    1.3.11                   py37_0
cryptography              2.7              py37h1ba5d50_0
cudatoolkit               10.0.130                      0
cycler                    0.10.0                   py37_0
dbus                      1.13.6               h746ee38_0
decorator                 4.4.0                    py37_1
expat                     2.2.6                he6710b0_0
fontconfig                2.13.0               h9420a91_0
freetype                  2.9.1                h8a8886c_1
future                    0.17.1                py37_1000    conda-forge
glib                      2.56.2               hd408876_0
gst-plugins-base          1.14.0               hbbd80ab_1
gstreamer                 1.14.0               hb453b48_1
icu                       58.2                 h9c2bf20_1
idna                      2.8                      py37_0
intel-openmp              2019.4                      243
ipython                   7.7.0            py37h39e3cac_0
ipython_genutils          0.2.0                    py37_0
jedi                      0.13.3                   py37_0
jpeg                      9b                   h024ee3a_2
kiwisolver                1.1.0            py37he6710b0_0
libarchive                3.3.3                h5d8350f_5
libedit                   3.1.20181209         hc058e9b_0
libffi                    3.2.1                hd88cf55_4
libgcc-ng                 9.1.0                hdf63c60_0
libgfortran-ng            7.3.0                hdf63c60_0
libpng                    1.6.37               hbc83047_0
libstdcxx-ng              9.1.0                hdf63c60_0
libtiff                   4.0.10            h2733197_1001
libuuid                   1.0.3                h1bed415_2
libxcb                    1.13                 h1bed415_1
libxml2                   2.9.9                hea5a465_1
libxslt                   1.1.33               h7d1a2b0_0
lxml                      4.4.0            py37h7ec2d77_0    conda-forge
lz4-c                     1.8.1.2              h14c3975_0
lzo                       2.10                 h49e0be7_2
matplotlib                3.1.0            py37h5429711_0
mkl                       2019.4                      243
mkl-service               2.0.2            py37h7b6447c_0
mkl_fft                   1.0.12           py37ha843d7b_0
mkl_random                1.0.2            py37hd81dba3_0
ncurses                   6.1                  he6710b0_1
ninja                     1.9.0            py37hfd86e86_0
numpy                     1.16.4           py37h7e9f1db_0
numpy-base                1.16.4           py37hde5b4d6_0
obspy                     1.1.0           py37h3010b51_1002    conda-forge
olefile                   0.46                     py37_0
openssl                   1.1.1c               h7b6447c_1
parso                     0.5.0                      py_0
pcre                      8.43                 he6710b0_0
pexpect                   4.7.0                    py37_0
pickleshare               0.7.5                    py37_0
pillow                    6.1.0            py37h34e0f95_0
pip                       19.1.1                   py37_0
prompt_toolkit            2.0.9                    py37_0
ptyprocess                0.6.0                    py37_0
pycosat                   0.6.3            py37h14c3975_0
pycparser                 2.19                     py37_0
pygments                  2.4.2                      py_0
pyopenssl                 19.0.0                   py37_0
pyparsing                 2.4.0                      py_0
pyqt                      5.9.2            py37h05f1152_2
pysocks                   1.7.0                    py37_0
python                    3.7.4                h265db76_0
python-dateutil           2.8.0                    py37_0
python-libarchive-c       2.8                     py37_13
pytorch                   1.2.0           py3.7_cuda10.0.130_cudnn7.6.2_0    pytorch
pytz                      2019.1                     py_0
qt                        5.9.7                h5867ecd_1
readline                  7.0                  h7b6447c_5
requests                  2.22.0                   py37_0
ruamel_yaml               0.15.46          py37h14c3975_0
scipy                     1.3.0            py37h7c811a0_0
setuptools                41.0.1                   py37_0
sip                       4.19.8           py37hf484d3e_0
six                       1.12.0                   py37_0
sqlalchemy                1.3.6            py37h516909a_0    conda-forge
sqlite                    3.29.0               h7b6447c_0
tk                        8.6.8                hbc83047_0
torchvision               0.4.0                py37_cu100    pytorch
tornado                   6.0.3            py37h7b6447c_0
tqdm                      4.32.1                     py_0
traitlets                 4.3.2                    py37_0
urllib3                   1.24.2                   py37_0
vision                    0.0.3                     dev_0    <develop>
wcwidth                   0.1.7                    py37_0
wheel                     0.33.4                   py37_0
xz                        5.2.4                h14c3975_4
yaml                      0.1.7                had09818_2
yews                      0.0.5                     dev_0    <develop>
zlib                      1.2.11               h7b6447c_3
zstd                      1.3.7                h0b5b093_0
```

**Comment by QuLogic** [comment]
This issue is about Ubuntu packages, not conda. You should report problems with the conda-forge packages on their tracker https://github.com/conda-forge/obspy-feedstock/issues

**Comment by lijunzh** [comment]
@QuLogic Thanks for the advice. However, the problem shows up only after I updated the Ubuntu via "apt upgrade". Thus, I suspect it breaks some soft-link during that update. Since the error message are similar here, I figured it will be helpful to know how it was fixed previously. 

In fact, I only need the `obspy.read` working. I have a strip-off version of that for SAC that I can work around this issue. But if figure it will be informative to report this issue on GitHub.
